File: лабы/лаба2.py
#include <stdio.h>

int main() {
    // PART 1: Matrix 3x3 (double)
    double matrix3x3[3][3];
    double main_diag_sum = 0, secondary_diag_sum = 0;
    int i, j;

    printf("Enter elements of 3x3 matrix (double):\n");
    for (i = 0; i < 3; i++) {
        for (j = 0; j < 3; j++) {
            printf("matrix3x3[%d][%d] = ", i, j);
            scanf("%lf", &matrix3x3[i][j]);
        }
    }

    
    for (i = 0; i < 3; i++) {
        main_diag_sum += matrix3x3[i][i];              
        secondary_diag_sum += matrix3x3[i][2 - i];     
    }

    printf("\nSum of main diagonal: %.2f\n", main_diag_sum);
    printf("Sum of secondary diagonal: %.2f\n", secondary_diag_sum);

    // PART 2: Matrix 2x2 (int) and its square
    int matrix2x2[2][2];
    int result[2][2];

    printf("\nEnter elements of 2x2 matrix (int):\n");
    for (i = 0; i < 2; i++) {
        for (j = 0; j < 2; j++) {
            printf("matrix2x2[%d][%d] = ", i, j);
            scanf("%d", &matrix2x2[i][j]);
        }
    }

    
    result[0][0] = matrix2x2[0][0] * matrix2x2[0][0] + matrix2x2[0][1] * matrix2x2[1][0];
    result[0][1] = matrix2x2[0][0] * matrix2x2[0][1] + matrix2x2[0][1] * matrix2x2[1][1];
    result[1][0] = matrix2x2[1][0] * matrix2x2[0][0] + matrix2x2[1][1] * matrix2x2[1][0];
    result[1][1] = matrix2x2[1][0] * matrix2x2[0][1] + matrix2x2[1][1] * matrix2x2[1][1];

    printf("\nSquare of the matrix:\n");
    printf("%d %d\n", result[0][0], result[0][1]);
    printf("%d %d\n", result[1][0], result[1][1]);

    return 0;
}

File: лабы/лаба2.c
#include <stdio.h>

int main() {
   	char choice = 'y';
   	while (choice == 'y' || choice == 'Y') {
        printf("\n=== Solving Quadratic Equation ===\n");
        
        int n = 4;
		double mat[n][n];
	    double sum_main = 0, sum_sec = 0;
	    int i, j;
	    int mat2[2][2];
	    int res[2][2];
		
	    printf("Enter elements of %dx%d matrix (double):\n", n, n);
	    for (i = 0; i < n; i++) {
	        for (j = 0; j < n; j++) {
	            printf("mat[%d][%d] = ", i, j);
	            scanf("%lf", &mat[i][j]);
	    }
	    }
	
	    for (i = 0; i < n; i++) {
	        sum_main += mat[i][i];
	        sum_sec += mat[i][n - 1 - i];
	    }
	
	    printf("\nSum of main diagonal: %.2f\n", sum_main);
	    printf("Sum of secondary diagonal: %.2f\n", sum_sec);
	
	    printf("\nEnter elements of 2x2 matrix (int):\n");
	    for (i = 0; i < 2; i++) {
	        for (j = 0; j < 2; j++) {
	            printf("mat2[%d][%d] = ", i, j);
	            scanf("%d", &mat2[i][j]);
	        }
	    }
	
	    res[0][0] = mat2[0][0] * mat2[0][0] + mat2[0][1] * mat2[1][0];
	    res[0][1] = mat2[0][0] * mat2[0][1] + mat2[0][1] * mat2[1][1];
	    res[1][0] = mat2[1][0] * mat2[0][0] + mat2[1][1] * mat2[1][0];
	    res[1][1] = mat2[1][0] * mat2[0][1] + mat2[1][1] * mat2[1][1];
	
	    printf("\nSquare of the matrix:\n");
	    printf("%d %d\n", res[0][0], res[0][1]);
	    printf("%d %d\n", res[1][0], res[1][1]);
		printf("continieu - 1,stop-0");
		 printf("\nDo you want to solve another equation? (y/n): ");
        scanf(" %c", &choice); 
	}
    return 0;
}
